go/arrow/compute/arithmetic.go

```diff
@@ -627,6 +627,8 @@ func RegisterScalarArithmetic(reg FunctionRegistry) {
 	}{
 		{"sub_unchecked", kernels.OpSub, decPromoteAdd, subUncheckedDoc},
 		{"sub", kernels.OpSubChecked, decPromoteAdd, subDoc},
+		{"subtract_unchecked", kernels.OpSub, decPromoteAdd, subUncheckedDoc},
+		{"subtract", kernels.OpSubChecked, decPromoteAdd, subDoc},
```

Comment:
just made things easier since for all of the *other* functions I was able to have a simple loop since the names matched and I didn't want to have a separate case for only subtract. It's not strictly needed, it just made some code cleaner.